packages/core/admin/webpack.config.dev.js

```diff
@@ -47,17 +47,32 @@ module.exports = () => {
 
   const config = webpackConfig(args);
 
-  if (analyzeBundle) {
-    config.plugins.push(new BundleAnalyzerPlugin());
-  }
-
-  if (analyzeDuplicateDependencies === 'true') {
-    config.plugins.push(new DuplicateReporterPlugin());
-  }
-
   return {
     ...config,
 
+    resolve: {
+      ...config.resolve,
+      alias: {
+        ...config.resolve.alias,
+        /**
+         * These aliases mean that we're actually looking at the source of the code
+         * as opposed to the compiled version of the code. This is useful for a better local DX.
+         */
+        ...plugins.reduce((acc, plugin) => {
+          acc[`${plugin.name}/strapi-admin`] = path.join(plugin.directory, 'admin', 'src');
+
+          return acc;
+        }, {}),
+      },
+    },
+
+    plugins: [
+      ...config.plugins,
+      analyzeBundle && new BundleAnalyzerPlugin(),
+      analyzeDuplicateDependencies === 'true' && new DuplicateReporterPlugin(),
+      process.env.REACT_REFRESH !== 'false' && new ReactRefreshWebpackPlugin(),
```

Comment:
Why did you move that to the `dev` config? Are plugin developers able to still use it?

Reply:
ahhh is that why it's there for the `watchAdmin` stuff, I can put this back then, thanks 🧠